### ISSUE TITLE
🚀 Feature: Allow default plugins to have default configuration

### ISSUE THREAD

**Issue by asos-robbell** [backstage/backstage#20610]
### 🔖 Feature description

PR https://github.com/backstage/backstage/pull/18699 enabled passing a `string[]` of defaultPlugins to MkDocs' configuration but, as it's only a `string[]`, configuration of those plugins isn't accepted.

[Several](https://github.com/backstage/backstage/pull/18830#issuecomment-1757411813) [users](https://github.com/backstage/backstage/pull/18699#issuecomment-1700964047) have requested this feature, particularly around configuration of Kroki's `ServerUrl` and `FencePrefix`. Requiring a user to set this configuration themselves means the set of defaultPlugins is cleared and they need to re-add each plugin, regardless of whether they're passing configuration for it.

### 🎤 Context

Below is a common example of how most users expect this feature to work—as well as providing a list of plugins, they provide their configuration too. This is useful when you want consistency across an organisation, for example, ensuring that all users use a known endpoint when configuring a plugin:

``` yaml
defaultPlugins:
  - techdocs-core
  - awesome-pages
  - kroki:
      FencePrefix: ''
      ServerUrl: 'https://myprivatekrokiserver.com'
  - myOtherPlugin:
      MyConfig: '...'
```

### ✌️ Possible Implementation

Allowing defaultPlugins to accept an `object[]` instead of a `string[]`.

A default `mkdocs` file [has been discussed too](https://github.com/backstage/backstage/pull/18830), although I think most users would still need to set their structure, site title etc. so unless this uses some form of inheritance or extensibility, I don't think this approach will work.

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by krunalhinguu** [comment]
Hi @tudi2d @asos-robbell is this issue is still open?

if yes, i would like to work on this issue.

**Comment by asos-robbell** [comment]
> Hi @tudi2d @asos-robbell is this issue is still open?
> 
> if yes, i would like to work on this issue.

Hi @krunalhinguu, yes this is till open. Let me know if you'd like any assistance.

**Comment by krunalhinguu** [comment]
@asos-robbell do you have some more information related to this change. Incase you have some existing approach for this change we can discuss that too

**Comment by asos-robbell** [comment]
Hi @krunalhinguu, it looks as though the feature to add a string[] of plugins (https://github.com/backstage/backstage/pull/18699) was influenced by the CLI's capabilities—the [CLI only accepts a string array](https://backstage.io/docs/features/techdocs/cli/) too.

I've not thought about the plumbing required to enable default configuration for plugins, so all ideas welcome.

**Comment by krunalhinguu** [comment]
perfect. Thanks @asos-robbell 

One another question, could you please guide me with how i can replicate this scenario or possible scenario which i can test once the work is done.

**Comment by krunalhinguu** [comment]
As per the approach I followed, I've made changes to the way we handle default plugins. Instead of using simple names (like strings), I've tried to accommodate detailed objects. I've ensured that these changes don't disrupt the existing system; everything still works as it did before. This new approach is akin to following a blueprint, ensuring that our system remains smooth and ready for future enhancements.
